### PR TITLE
Reland python ops

### PR DESCRIPTION
**Pull request by kumpera**
Waiting for the revert to land.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/distributed/_functional_collectives.py [hunk 1]
```diff
@@ -15,6 +15,17 @@
...
 
+"""
+The old check `sys.executable == 'torch_deploy'` for torch::deploy was replaced here:
+https://github.com/pytorch/multipy/pull/138/files#diff-dae4e20139ff6af007a16cc6888d0e3c1d40d297cb7aef89d8e6cc201caacb9eR124
+
+The new check is a bit more hackish, but that's all we have for now.
+
+"""
+def _is_running_under_torch_deploy():
+    return torch._meta_registrations is object
```
Comment (albanD) [inline torch/distributed/_functional_collectives.py:26]: Could you send a PR to multipy to add a comment next to that line saying that we rely on this particular "object" state and it should not be modified?

**Review by albanD — APPROVED** [review]
Pretty hacky but I guess we don't have a choice...

File: torch/distributed/_functional_collectives.py (hunk 1 above)
Comment (kumpera) [reply torch/distributed/_functional_collectives.py:26]: https://github.com/kumpera/multipy/pull/1

Comment (kumpera) [reply torch/distributed/_functional_collectives.py:26]: https://github.com/pytorch/multipy/pull/317

**Comment by DanilBaibak** [comment]
@kumpera sorry, I have to revert your PR. It leads to internal failures. Here you can find more details - [D45066425](https://www.internalfb.com/diff/D45066425).

**Pull request closed without merging**